SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -413,36 +413,54 @@
          
        We want to run after the core "ResolveCodeAnalysisRuleSet" target which
        assigns a value to "ResolvedCodeAnalysisRuleSet".
-  -->
+       
+       Conditions:
+       * project is a test project -> turn off Roslyn analysis
+       * project is excluded -> turn off Roslyn analysis
+       * downloaded SonarLint ruleset and analyzer found -> add SonarLint config to existing analysis config
+       * downloaded SonarLint ruleset and analyzer not found -> no change to existing config
+   -->
   <Target Name="OverrideRoslynCodeAnalysisProperties"
         Condition=" $(SonarQubeTempPath) != '' "
         DependsOnTargets="SonarQubeCategoriseProject"
         AfterTargets="ResolveCodeAnalysisRuleSet"
         BeforeTargets="CoreCompile">
 
+    <!-- Check whether to turn off code analysis -->
     <PropertyGroup>
-      <!-- Check if the ruleset exists -->
-      <SonarQubeRoslynRulesetFullName>$(SonarQubeConfigPath)SonarQubeRoslyn-cs.ruleset</SonarQubeRoslynRulesetFullName>
-      <SonarQubeRoslynRulesetExists>$([System.IO.File]::Exists($(SonarQubeRoslynRulesetFullName)))</SonarQubeRoslynRulesetExists>
-
-      <!-- Run FxCop only if the ruleset exists (else it means no Roslyn rules are enabled) -->
-      <SonarQubeRunRoslynCodeAnalysis>$(SonarQubeRoslynRulesetExists)</SonarQubeRunRoslynCodeAnalysis>
-
-      <!-- For the time being only run for C# -->
-      <SonarQubeRunRoslynCodeAnalysis Condition=" $(Language) != 'C#' ">false</SonarQubeRunRoslynCodeAnalysis>
-
-      <!-- Don't run FxCop if the project is excluded -->
-      <SonarQubeRunRoslynCodeAnalysis Condition=" $(SonarQubeExclude) == 'true' ">false</SonarQubeRunRoslynCodeAnalysis>
-
-      <!-- Don't run FxCop if the project is a test one -->
-      <SonarQubeRunRoslynCodeAnalysis Condition=" $(SonarQubeTestProject) == 'true' ">false</SonarQubeRunRoslynCodeAnalysis>
-
+      <SonarQubeDisableRoslynCodeAnalysis Condition="$(SonarQubeExclude) == 'true' OR $(SonarQubeTestProject) == 'true' ">true</SonarQubeDisableRoslynCodeAnalysis>
     </PropertyGroup>
-
-    <PropertyGroup Condition=" $(SonarQubeRunRoslynCodeAnalysis) != 'true' ">
+    
+    <PropertyGroup Condition=" $(SonarQubeDisableRoslynCodeAnalysis) == 'true' ">
       <ErrorLog></ErrorLog>
       <ResolvedCodeAnalysisRuleSet></ResolvedCodeAnalysisRuleSet>
+    </PropertyGroup>
+    <!-- else -->
+    <CallTarget Targets="SetRoslynCodeAnalysisProperties"
+                Condition="$(SonarQubeDisableRoslynCodeAnalysis) != 'true'"/>
```

Comment:
I moved the "set property" logic into a separate target that is invoked conditionally so I didn't have to keep checking the "is disabled" condition at each stage.